### PR TITLE
[GLUTEN-7220][CH]Fix expand bug in grouping sets query

### PR DESCRIPTION
**Pull request by KevinyhZou**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#7220)

## How was this patch tested?
UT

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: cpp-ch/local-engine/Operator/ExpandTransform.cpp [hunk 1]
```diff
@@ -83,56 +84,53 @@ ExpandTransform::Status ExpandTransform::prepare()
...
     const auto & input_header = getInputs().front().getHeader();
     const auto & input_columns = input_chunk.getColumns();
     const auto & types = project_set_exprs.getTypes();
-    const auto & kinds = project_set_exprs.getKinds()[expand_expr_iterator];
-    const auto & fields = project_set_exprs.getFields()[expand_expr_iterator];
     size_t rows = input_chunk.getNumRows();
 
-    DB::Columns columns(types.size());
-    for (size_t col_i = 0; col_i < types.size(); ++col_i)
+    DB::MutableColumns columns(types.size());
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Operator/ExpandTransform.cpp:101]: 这里为什么要生成新的columns,   有没有可能利用复用老的column,  这个操作相比原来要重不少。

File: cpp-ch/local-engine/Operator/ExpandTransorm.h [hunk 2]
```diff
@@ -46,7 +46,6 @@ class ExpandTransform : public DB::IProcessor
     ExpandField project_set_exprs;
     bool has_input = false;
     bool has_output = false;
-    size_t expand_expr_iterator = 0;
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Operator/ExpandTransorm.h:49]: 原来的expand流式输出去掉了吗？ @lgbo-ustc 看下这里是否ok ?

**Review by taiyang-li — CHANGES_REQUESTED** [review]

**Comment by lgbo-ustc** [comment]
let's rename the header file in this PR
image

File: cpp-ch/local-engine/Operator/ExpandTransform.cpp (hunk 1 above)
Comment (zhanglistar) [reply cpp-ch/local-engine/Operator/ExpandTransform.cpp:101]: 这里确实有点重了

**Comment by lgbo-ustc** [comment]
> let's rename the header file in this PR image

@KevinyhZou

**Review by taiyang-li — DISMISSED** [review]
LGTM

**Comment by KevinyhZou** [comment]
> > let's rename the header file in this PR image
> 
> @KevinyhZou The header file name is typo

done @lgbo-ustc

**Review by lgbo-ustc — APPROVED** [review]

**Pull request merged**